### DISCUSSION TITLE
cross-layer effects

### DISCUSSION THREAD

**Discussion by SimonMontfort** [statnet/ergm.multi#34] [Q&A]
I am getting acquainted with ergm.multi.. I am wondering if it is possible to model endogenous cross-layer effects with a multi-ergm. Specifically, I am interested in modelling positive and negative ties endogenously (which could be conceptualised as two networks). If there is already a negative tie, can we model the probability of forming a positive tie in any way?

I've tried the following: 

```
# Load ergm and ergm.multi packages
library(ergm)
library(ergm.multi)

# Construct the two layers' networks (from your original code):
nw12 <- network.initialize(20, dir = FALSE, bipartite = 5)
nw34 <- network.initialize(20, dir = FALSE, bipartite = 5)
nw12 %v% "mode" <- rep(1:2, c(5, 15))
nw34 %v% "mode" <- rep(1:2, c(5, 15))

set.seed(1)
# Assign random ties for testing
nw12[,] <- sample(c(1, 0), 20, replace = TRUE)
nw34[,] <- sample(c(1, 0), 20, replace = TRUE)

# Assign tie types
nw12 %e% "tie_type" <- rep("positive", nrow(as.edgelist(nw12)))
nw34 %e% "tie_type" <- rep("negative", nrow(as.edgelist(nw34)))

res <- ergm(
  Layer(nw12, nw34) ~ 
    L(~ edges + edgecov(nw34, attrname="tie_type"), ~`1`) 
    + L(~ edges + edgecov(nw12, attrname="tie_type"), ~`2`) 
    # + L(~ edges, ~ `2` | `1`)  
)
```

Would be interested to hear thoughts on this problem.

**Comment by krivit** [comment]
A few thoughts:
1. Loading `ergm.multi` automatically loads `ergm`.
2. `"mode"` attribute is only needed if you have a scenario in which layers have heterogeneous "partedness".
3. `edgecov()` of another layer treats that layer as exogenous; this can be viewed as modelling the conditional probability of a "positive" tie given a "negative" tie or given its absence. Endogenous modelling doesn't really let you specify direction, at least not when you only observe one network: you can only model the joint distribution of "positive" and "negative" ties. If that's what you want, something like this might do:

```{r}
library(ergm.multi)

# Construct the two layers' networks (from your original code):
nw12 <- network.initialize(20, dir = FALSE, bipartite = 5)
nw34 <- network.initialize(20, dir = FALSE, bipartite = 5)
# mode is unnecessary unless you have layers with heterogeneous modes.

set.seed(1)
# Assign random ties for testing
nw12[,] <- sample(c(1, 0), 20, replace = TRUE)
nw34[,] <- sample(c(1, 0), 20, replace = TRUE)

# There is no need to assign tie types: one network will have positive and the other negative ties

res <- ergm(
  Layer(nw12, nw34) ~ # nw12 as first layer, n34 as second layer
    L(~ edges, ~`1`) # edgecov treats ties of the other layer as exogenous
  + L(~ edges, ~`2`)  
  + L(~ edges, ~ `2` & `1`) # co-occurrence of positive and negative ties
)

summary(res)
```
